Source/diablo.cpp

```diff
@@ -289,13 +289,19 @@ void diablo_splash()
 	UiTitleDialog();
 }
 
+void diablo_terminate()
+{
+	init_cleanup();
+	SaveGamma();
+	SDL_Quit();
+}
+
 int APIENTRY WinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, LPSTR lpCmdLine, int nCmdShow)
 {
+	atexit(diablo_terminate);

```

Comment:
At exit is not equally well supported on all platforms so we should probably try to limit our reliance on it

Reply:
I don't know about old systems but on macOS SaveGamma() not called if you close window.
And only way i found is call SaveGamma() from at exit.
And what mean " try to limit our reliance on it"?
I move at exit from `SourceX/miniwin/misc.cpp` to `Source/diablo.cpp`.

Reply:
Since Source comes from upstream we also try and make a minimal amount of changes there so that we can more easily merge updates from upstream. Instead we prefer making changes in SourceX. What you mean by "old systems"?

Reply:
"old systems" - platforms without support for at exit. Sorry my english)
Ok, I try move code `main.cpp`.

Reply:
Ok, people don't usually consider Android an old system :)
https://hg.libsdl.org/SDL/file/tip/docs/README-android.md#l431
But maybe I'm not understanding it correctly and `atexit` is fine, but `exit` is not?

Reply:
`Don't call exit() as it stops the activity badly.`
Function exit() used 10 times in code)
Yes, macOS receive a SDL_QUIT event if i close window.
But this event handled 7 times in upstream code.
I don't have android and can't test this but i think saving don't work on android.
And for android users nothing changed. Maybe later android developer fix this.

Reply:
Well, I would like to find a solution that doesn't result in a problem for other platforms. Afaik load/save works for the Android port. Imagin they have the same opinion about macOS and just break things for other platofrms then we just have a mess of a siguation where things randomly stop working with each change to the code.